### PR TITLE
fix: adjustments to improove and fix some behaviors at pagination

### PR DESCRIPTION
**Pull request by wermesonsl**
## Description

When using pagination component in a project, through the SmatTable component, some behaviors  that need to be resolved were noticed:

- When selecting a number of items per page, the option selector appears broken in the layout;
- Also, it has a strange behavior over the table, causing the table footer to jitter slightly when I open the selector.

## Expected Behavior

- That selector for the number of items per page is completely visible;
- That selector is usable, even when placed at the bottom of a page, opening the options upwards.

## Screenshots

Captura de tela de 2023-03-28 16-13-41

## Additional Information

There are two other notes about pagination involving the table, so I'll leave them on record as another issue to resolve, leaving them here for informational purposes only:

- When pagination receives 0 as the total items value, it disappears;
- When using paging, the table is returning two Output emissions, causing the page to call data listing route twice instead of once.

### CODE REVIEW

**Review by mizek1 — APPROVED** [review]

File: projects/ion/src/lib/table/table.component.scss [hunk 1]
```diff
@@ -137,6 +138,7 @@ table {
   justify-content: space-between;
   padding: 16px;
   gap: spacing(2);
+  z-index: 900;
```
Comment (AlyssonMascarenhas) [inline projects/ion/src/lib/table/table.component.scss:141]: won't this be a problem when opening a modal?
I suggest creating a pattern for the z-index to solve this problem.

**Review by iurynogueira — COMMENTED** [review]
Waiting response to @AlyssonMascarenhas

File: projects/ion/src/lib/table/table.component.scss (hunk 1 above)
Comment (wermesonsl) [reply projects/ion/src/lib/table/table.component.scss:141]: That's a great suggestion. Imma work on this right now.

Comment (wermesonsl) [reply projects/ion/src/lib/table/table.component.scss:144]: Already done!

**Pull request merged**